influxql/statement_rewriter.go

```diff
@@ -149,3 +151,88 @@ func rewriteShowTagKeysStatement(stmt *ShowTagKeysStatement) (Statement, error)
 		Dedupe:     true,
 	}, nil
 }
+
+func rewriteShowTagValuesStatement(stmt *ShowTagValuesStatement) (Statement, error) {
+	// Check for time in WHERE clause (not supported).
+	if HasTimeExpr(stmt.Condition) {
+		return nil, errors.New("SHOW TAG VALUES doesn't support time in WHERE clause")
+	}
+
+	condition := stmt.Condition
+	if len(stmt.Sources) > 0 {
+		if source, ok := stmt.Sources[0].(*Measurement); ok {
+			var expr Expr
+			if source.Regex != nil {
+				expr = &BinaryExpr{
+					Op:  EQREGEX,
+					LHS: &VarRef{Val: "name"},
+					RHS: &RegexLiteral{Val: source.Regex.Val},
+				}
+			} else if source.Name != "" {
+				expr = &BinaryExpr{
+					Op:  EQ,
+					LHS: &VarRef{Val: "name"},
+					RHS: &StringLiteral{Val: source.Name},
+				}
+			}
+
+			// Set condition or "AND" together.
+			if condition == nil {
+				condition = expr
+			} else {
+				condition = &BinaryExpr{Op: AND, LHS: expr, RHS: condition}
+			}
+		}
+	}
```

Comment:
I refactored this section out in my `SHOW SERIES` branch. You can pull it out if you want to reuse it:

https://github.com/benbjohnson/influxdb/blob/5997efa5fbbf000d17e8a5e81782d9ff2b2d72ec/influxql/statement_rewriter.go#L123-L172